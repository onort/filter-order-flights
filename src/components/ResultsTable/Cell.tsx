import React from "react"
import cx from "classnames"

import styles from "./ResultsTable.module.scss"

interface Props {
  className?: string
}

const Cell: React.SFC<Props> = props => {
  const classNames = cx(styles.cell, props.className)

  return <div className={classNames}>{props.children}</div>
}

export default Cell
