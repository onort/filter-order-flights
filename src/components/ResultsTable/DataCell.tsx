import React from "react"
import cx from "classnames"

import styles from "./ResultsTable.module.scss"
import { Cell } from "./"

interface Props {
  className?: string
  sub: string
  val: string
}

const DataCell: React.SFC<Props> = props => {
  const { sub, val } = props
  const classNames = cx(styles.dataCell, props.className)
  return (
    <Cell className={classNames}>
      <span className={styles.val}>{val}</span>
      <span className={styles.sub}>{sub}</span>
    </Cell>
  )
}

export default DataCell
