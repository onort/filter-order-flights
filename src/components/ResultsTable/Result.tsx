import React, { Component } from "react"
import cx from "classnames"
import dayjs from "dayjs"

import styles from "./ResultsTable.module.scss"
import { Itinerary } from "../../types"
import { Button, ResultDetails } from "../"
import { Cell, DataCell, Row } from "./"
import { minsToString } from "../../utils"

interface State {
  showDetails: boolean
}

interface Props {
  className?: string
  itinerary: Itinerary
}

class Result extends Component<Props, State> {
  public state = { showDetails: false }

  public toggleDetails = (e: React.MouseEvent<HTMLElement>) =>
    this.setState(curState => ({ showDetails: !curState.showDetails }))

  public render() {
    const classNames = cx(
      styles.row,
      { [styles.active]: this.state.showDetails },
      this.props.className
    )
    const {
      carriers,
      destination,
      duration,
      legs,
      lowestPrice,
      origin,
      pricingOptions,
      stops
    } = this.props.itinerary

    return (
      <>
        <Row className={classNames}>
          <Cell className={styles.carrierCell}>
            {carriers.length === 1 ? (
              <img
                className={styles.logo}
                src={carriers[0].imgUrl}
                alt={`${carriers[0].name} Logo`}
              />
            ) : (
              <>
                {carriers.map(carrier => (
                  <span key={carrier.id} className={styles.carrierName}>
                    {carrier.name}
                  </span>
                ))}
              </>
            )}
          </Cell>
          <DataCell
            sub={origin.airportCode}
            val={dayjs(origin.time).format("HH:mm")}
          />
          <DataCell
            sub={destination.airportCode}
            val={dayjs(destination.time).format("HH:mm")}
          />
          <DataCell
            sub={minsToString(duration)}
            val={stops === 0 ? "Direkt" : `${stops} Aktarma`}
          />
          <Cell className={styles.priceCell}>
            <div className={styles.price}>
              <div className={styles.dataCell}>
                <span className={styles.val}>{lowestPrice} TL</span>
                {carriers.length === 1 ? (
                  <span className={styles.sub}>{carriers[0].name}</span>
                ) : (
                  <span className={styles.sub}>Farklı havayolları</span>
                )}
              </div>
              <div className={styles.buttonContainer}>
                <Button
                  className={styles.viewButton}
                  text="İncele"
                  onClick={this.toggleDetails}
                />
              </div>
            </div>
          </Cell>
        </Row>
        {this.state.showDetails && (
          <ResultDetails legs={legs} pricingOptions={pricingOptions} />
        )}
      </>
    )
  }
}

export default Result
