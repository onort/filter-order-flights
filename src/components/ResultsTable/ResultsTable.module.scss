/* stylelint-disable declaration-colon-newline-after */
@import "../../styles/main.scss";

$priceCellPaddingHorizontal: map-get($spacers, small);
$priceCellPaddingVertical: map-get($spacers, normal);

.container {
  display: grid;
}

.row {
  display: grid;
  padding: map-get($spacers, small) map-get($spacers, normal);
  border-bottom: 1px solid $muted-border;
  grid-template-columns: repeat(4, 1fr) 2fr;

  &.header {
    display: grid;
    padding: 0 map-get($spacers, small);
    border-bottom: 1px solid $muted-border;
    color: $grey2;
    font-family: $font-family-primary;
    font-size: map-get($heading-sizes, h5);
    font-weight: 600;
    grid-template-columns: repeat(6, 1fr);
  }

  &.active {
    border-bottom: none;
    background-color: $element-bg;
  }
}

.cell {
  padding: map-get($spacers, large) map-get($spacers, normal);

  &.priceCell {
    padding:
      calc(#{map-get($spacers, large)} - #{$priceCellPaddingHorizontal})
      calc(#{map-get($spacers, normal)} - #{$priceCellPaddingHorizontal});
  }
}

.dataCell,
.carrierCell {
  display: flex;
  flex-direction: column;
  justify-content: center;

  span {
    display: block;
  }
}

.carrierName {
  color: $grey2;
  font-family: $font-family-primary;
  font-size: map-get($heading-sizes, h5);
  font-weight: 400;
}

.logo {
  max-width: 100%;
  height: auto;
}

.val {
  color: $black;
  font-family: $font-family-primary;
  font-size: map-get($font-sizes, large);
  font-weight: 600;
}

.sub {
  color: $grey2;
  font-family: $font-family-primary;
  font-size: map-get($heading-sizes, h5);
}

.price {
  display: flex;
  justify-content: space-between;
  padding: $priceCellPaddingVertical $priceCellPaddingHorizontal;
  background-color: $element-bg;
  border-radius: px-to-rem(2px);
}

.viewButton {
  background-color: $dark-blue;
}

.noResult {
  padding: map-get($spacers, x-large) map-get($spacers, large);
  color: $grey3;
  font-family: $font-family-primary;
  font-size: map-get($heading-sizes, h3);
  font-weight: 400;
}
