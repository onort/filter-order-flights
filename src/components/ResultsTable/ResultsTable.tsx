import React from "react"

import styles from "./ResultsTable.module.scss"
import { Itinerary } from "../../types"
import { Cell, Result, Row } from "./"

interface Props {
  data: Itinerary[]
}

const ResultsTable: React.SFC<Props> = props => {
  return (
    <div className={styles.container}>
      <Row className={styles.header}>
        <Cell>Havayolu</Cell>
        <Cell>Kalkış</Cell>
        <Cell>Varış</Cell>
        <Cell>Aktarma / Süre</Cell>
        <Cell>Fiyat</Cell>
        <Cell>Satın Al</Cell>
      </Row>
      {props.data.map(itinerary => (
        <Result key={itinerary.id} itinerary={itinerary} />
      ))}
    </div>
  )
}

export default ResultsTable
