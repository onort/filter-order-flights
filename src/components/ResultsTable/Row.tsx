import React from "react"
import cx from "classnames"

import styles from "./ResultsTable.module.scss"

interface Props {
  className?: string
}

const Row: React.SFC<Props> = props => {
  const classNames = cx(styles.row, props.className)

  return <div className={classNames}>{props.children}</div>
}

export default Row
