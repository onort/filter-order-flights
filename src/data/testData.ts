import { StopsType } from "../types"

export default [
  {
    carriers: [
      {
        code: "PC",
        displayCode: "PC",
        id: 1161,
        imgUrl: "http://s1.apideeplink.com/images/airlines/H9.png",
        name: "Pegasus Airlines"
      }
    ],

    destination: {
      airportCode: "ADB",
      time: "2017-04-14T09:10:00"
    },
    duration: 90,
    id: "9249-1704140740--31973-0-9250-1704140910",
    legs: [
      {
        arrival: {
          airportCode: "ADB",
          airportName: "Izmir",
          time: "2017-04-14T09:10:00"
        },
        carrier: "Pegasus Airlines",
        departure: {
          airportCode: "ADA",
          airportName: "Adana",
          time: "2017-04-14T07:40:00"
        },
        duration: 90,
        flightCode: "PC 3011"
      }
    ],
    lowestPrice: 66.99,
    origin: {
      airportCode: "ADA",
      time: "2017-04-14T07:40:00"
    },
    pricingOptions: [
      {
        agent: "Pegasus Airlines",
        price: 66.99
      }
    ],
    stops: 0,
    stopsType: StopsType.direct
  }
]

export const itineraryWithMultipleCarriers = {
  carriers: [
    {
      code: "PC",
      displayCode: "PC",
      id: 1161,
      imgUrl: "http://s1.apideeplink.com/images/airlines/H9.png",
      name: "Pegasus Airlines"
    },
    {
      code: "ABC",
      displayCode: "ABC",
      id: 123,
      imgUrl: "http://s1.apideeplink.com/images/airlines/H9.png",
      name: "ABC Airlines"
    }
  ],

  destination: {
    airportCode: "ADB",
    time: "2017-04-14T09:10:00"
  },
  duration: 90,
  id: "9249-1704140740--31973-0-9250-1704140910",
  legs: [
    {
      arrival: {
        airportCode: "ADB",
        airportName: "Izmir",
        time: "2017-04-14T09:10:00"
      },
      carrier: "Pegasus Airlines",
      departure: {
        airportCode: "ADA",
        airportName: "Adana",
        time: "2017-04-14T07:40:00"
      },
      duration: 90,
      flightCode: "PC 3011"
    }
  ],
  lowestPrice: 66.99,
  origin: {
    airportCode: "ADA",
    time: "2017-04-14T07:40:00"
  },
  pricingOptions: [
    {
      agent: "Pegasus Airlines",
      price: 66.99
    }
  ],
  stops: 0,
  stopsType: StopsType.direct
}
