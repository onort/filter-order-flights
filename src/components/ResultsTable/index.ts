import Cell from "./Cell"
import DataCell from "./DataCell"
import Result from "./Result"
import ResultsTable from "./ResultsTable"
import Row from "./Row"

export default ResultsTable

export { Cell, DataCell, Result, Row }
