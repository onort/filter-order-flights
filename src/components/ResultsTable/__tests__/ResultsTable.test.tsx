import React from "react"
import { shallow, mount } from "enzyme"

import ResultsTable from "../"
import testData from "../../../data/testData"

describe("<ResultsTable />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<ResultsTable data={testData} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have table header", () => {
    const wrapper = shallow(<ResultsTable data={testData} />)
    expect(wrapper.find(".header").exists()).toBe(true)
  })

  it("should not have a table header if data array is empty", () => {
    const wrapper = shallow(<ResultsTable data={[]} />)
    expect(wrapper.find(".header").exists()).toBe(false)
  })

  it("should show no result card if data array is empty", () => {
    const wrapper = shallow(<ResultsTable data={[]} />)
    expect(wrapper.find(".noResult").exists()).toBe(true)
  })

  it("should have correct amount of result items", () => {
    const wrapper = mount(<ResultsTable data={testData} />)
    expect(wrapper.find("Result")).toHaveLength(1)
    wrapper.unmount()
  })
})
