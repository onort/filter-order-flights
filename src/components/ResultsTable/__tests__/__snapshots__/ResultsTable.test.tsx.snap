// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsTable /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsTable
    data={
      Array [
        Object {
          "carriers": Array [
            Object {
              "code": "PC",
              "displayCode": "PC",
              "id": 1161,
              "imgUrl": "http://s1.apideeplink.com/images/airlines/H9.png",
              "name": "Pegasus Airlines",
            },
          ],
          "destination": Object {
            "airportCode": "ADB",
            "time": "2017-04-14T09:10:00",
          },
          "duration": 90,
          "id": "9249-1704140740--31973-0-9250-1704140910",
          "legs": Array [
            Object {
              "arrival": Object {
                "airportCode": "ADB",
                "airportName": "Izmir",
                "time": "2017-04-14T09:10:00",
              },
              "carrier": "Pegasus Airlines",
              "departure": Object {
                "airportCode": "ADA",
                "airportName": "Adana",
                "time": "2017-04-14T07:40:00",
              },
              "duration": 90,
              "flightCode": "PC 3011",
            },
          ],
          "lowestPrice": 66.99,
          "origin": Object {
            "airportCode": "ADA",
            "time": "2017-04-14T07:40:00",
          },
          "pricingOptions": Array [
            Object {
              "agent": "Pegasus Airlines",
              "price": 66.99,
            },
          ],
          "stops": 0,
          "stopsType": "direct",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          className="header"
        >
          <Cell>
            Havayolu
          </Cell>
          <Cell>
            Kalkış
          </Cell>
          <Cell>
            Varış
          </Cell>
          <Cell>
            Aktarma / Süre
          </Cell>
          <Cell>
            Fiyat
          </Cell>
          <Cell>
            Satın Al
          </Cell>
        </Row>,
        Array [
          <Result
            itinerary={
              Object {
                "carriers": Array [
                  Object {
                    "code": "PC",
                    "displayCode": "PC",
                    "id": 1161,
                    "imgUrl": "http://s1.apideeplink.com/images/airlines/H9.png",
                    "name": "Pegasus Airlines",
                  },
                ],
                "destination": Object {
                  "airportCode": "ADB",
                  "time": "2017-04-14T09:10:00",
                },
                "duration": 90,
                "id": "9249-1704140740--31973-0-9250-1704140910",
                "legs": Array [
                  Object {
                    "arrival": Object {
                      "airportCode": "ADB",
                      "airportName": "Izmir",
                      "time": "2017-04-14T09:10:00",
                    },
                    "carrier": "Pegasus Airlines",
                    "departure": Object {
                      "airportCode": "ADA",
                      "airportName": "Adana",
                      "time": "2017-04-14T07:40:00",
                    },
                    "duration": 90,
                    "flightCode": "PC 3011",
                  },
                ],
                "lowestPrice": 66.99,
                "origin": Object {
                  "airportCode": "ADA",
                  "time": "2017-04-14T07:40:00",
                },
                "pricingOptions": Array [
                  Object {
                    "agent": "Pegasus Airlines",
                    "price": 66.99,
                  },
                ],
                "stops": 0,
                "stopsType": "direct",
              }
            }
          />,
        ],
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Cell>
              Havayolu
            </Cell>,
            <Cell>
              Kalkış
            </Cell>,
            <Cell>
              Varış
            </Cell>,
            <Cell>
              Aktarma / Süre
            </Cell>,
            <Cell>
              Fiyat
            </Cell>,
            <Cell>
              Satın Al
            </Cell>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Havayolu",
            },
            "ref": null,
            "rendered": "Havayolu",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Kalkış",
            },
            "ref": null,
            "rendered": "Kalkış",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Varış",
            },
            "ref": null,
            "rendered": "Varış",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Aktarma / Süre",
            },
            "ref": null,
            "rendered": "Aktarma / Süre",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Fiyat",
            },
            "ref": null,
            "rendered": "Fiyat",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Satın Al",
            },
            "ref": null,
            "rendered": "Satın Al",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "9249-1704140740--31973-0-9250-1704140910",
        "nodeType": "class",
        "props": Object {
          "itinerary": Object {
            "carriers": Array [
              Object {
                "code": "PC",
                "displayCode": "PC",
                "id": 1161,
                "imgUrl": "http://s1.apideeplink.com/images/airlines/H9.png",
                "name": "Pegasus Airlines",
              },
            ],
            "destination": Object {
              "airportCode": "ADB",
              "time": "2017-04-14T09:10:00",
            },
            "duration": 90,
            "id": "9249-1704140740--31973-0-9250-1704140910",
            "legs": Array [
              Object {
                "arrival": Object {
                  "airportCode": "ADB",
                  "airportName": "Izmir",
                  "time": "2017-04-14T09:10:00",
                },
                "carrier": "Pegasus Airlines",
                "departure": Object {
                  "airportCode": "ADA",
                  "airportName": "Adana",
                  "time": "2017-04-14T07:40:00",
                },
                "duration": 90,
                "flightCode": "PC 3011",
              },
            ],
            "lowestPrice": 66.99,
            "origin": Object {
              "airportCode": "ADA",
              "time": "2017-04-14T07:40:00",
            },
            "pricingOptions": Array [
              Object {
                "agent": "Pegasus Airlines",
                "price": 66.99,
              },
            ],
            "stops": 0,
            "stopsType": "direct",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            className="header"
          >
            <Cell>
              Havayolu
            </Cell>
            <Cell>
              Kalkış
            </Cell>
            <Cell>
              Varış
            </Cell>
            <Cell>
              Aktarma / Süre
            </Cell>
            <Cell>
              Fiyat
            </Cell>
            <Cell>
              Satın Al
            </Cell>
          </Row>,
          Array [
            <Result
              itinerary={
                Object {
                  "carriers": Array [
                    Object {
                      "code": "PC",
                      "displayCode": "PC",
                      "id": 1161,
                      "imgUrl": "http://s1.apideeplink.com/images/airlines/H9.png",
                      "name": "Pegasus Airlines",
                    },
                  ],
                  "destination": Object {
                    "airportCode": "ADB",
                    "time": "2017-04-14T09:10:00",
                  },
                  "duration": 90,
                  "id": "9249-1704140740--31973-0-9250-1704140910",
                  "legs": Array [
                    Object {
                      "arrival": Object {
                        "airportCode": "ADB",
                        "airportName": "Izmir",
                        "time": "2017-04-14T09:10:00",
                      },
                      "carrier": "Pegasus Airlines",
                      "departure": Object {
                        "airportCode": "ADA",
                        "airportName": "Adana",
                        "time": "2017-04-14T07:40:00",
                      },
                      "duration": 90,
                      "flightCode": "PC 3011",
                    },
                  ],
                  "lowestPrice": 66.99,
                  "origin": Object {
                    "airportCode": "ADA",
                    "time": "2017-04-14T07:40:00",
                  },
                  "pricingOptions": Array [
                    Object {
                      "agent": "Pegasus Airlines",
                      "price": 66.99,
                    },
                  ],
                  "stops": 0,
                  "stopsType": "direct",
                }
              }
            />,
          ],
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Cell>
                Havayolu
              </Cell>,
              <Cell>
                Kalkış
              </Cell>,
              <Cell>
                Varış
              </Cell>,
              <Cell>
                Aktarma / Süre
              </Cell>,
              <Cell>
                Fiyat
              </Cell>,
              <Cell>
                Satın Al
              </Cell>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Havayolu",
              },
              "ref": null,
              "rendered": "Havayolu",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Kalkış",
              },
              "ref": null,
              "rendered": "Kalkış",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Varış",
              },
              "ref": null,
              "rendered": "Varış",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Aktarma / Süre",
              },
              "ref": null,
              "rendered": "Aktarma / Süre",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Fiyat",
              },
              "ref": null,
              "rendered": "Fiyat",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Satın Al",
              },
              "ref": null,
              "rendered": "Satın Al",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9249-1704140740--31973-0-9250-1704140910",
          "nodeType": "class",
          "props": Object {
            "itinerary": Object {
              "carriers": Array [
                Object {
                  "code": "PC",
                  "displayCode": "PC",
                  "id": 1161,
                  "imgUrl": "http://s1.apideeplink.com/images/airlines/H9.png",
                  "name": "Pegasus Airlines",
                },
              ],
              "destination": Object {
                "airportCode": "ADB",
                "time": "2017-04-14T09:10:00",
              },
              "duration": 90,
              "id": "9249-1704140740--31973-0-9250-1704140910",
              "legs": Array [
                Object {
                  "arrival": Object {
                    "airportCode": "ADB",
                    "airportName": "Izmir",
                    "time": "2017-04-14T09:10:00",
                  },
                  "carrier": "Pegasus Airlines",
                  "departure": Object {
                    "airportCode": "ADA",
                    "airportName": "Adana",
                    "time": "2017-04-14T07:40:00",
                  },
                  "duration": 90,
                  "flightCode": "PC 3011",
                },
              ],
              "lowestPrice": 66.99,
              "origin": Object {
                "airportCode": "ADA",
                "time": "2017-04-14T07:40:00",
              },
              "pricingOptions": Array [
                Object {
                  "agent": "Pegasus Airlines",
                  "price": 66.99,
                },
              ],
              "stops": 0,
              "stopsType": "direct",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<ResultsTable /> should match snapshot 2`] = `"<div class=\\"container\\"><div class=\\"row header\\"><div class=\\"cell\\">Havayolu</div><div class=\\"cell\\">Kalkış</div><div class=\\"cell\\">Varış</div><div class=\\"cell\\">Aktarma / Süre</div><div class=\\"cell\\">Fiyat</div><div class=\\"cell\\">Satın Al</div></div><div class=\\"row row\\"><div class=\\"cell carrierCell\\"><img class=\\"logo\\" src=\\"http://s1.apideeplink.com/images/airlines/H9.png\\" alt=\\"Pegasus Airlines Logo\\"/></div><div class=\\"cell dataCell origin\\"><span class=\\"val\\">07:40</span><span class=\\"sub\\">ADA</span><img class=\\"plane\\" src=\\"plane.svg\\" alt=\\"Plane Icon\\"/></div><div class=\\"cell dataCell\\"><span class=\\"val\\">09:10</span><span class=\\"sub\\">ADB</span></div><div class=\\"cell dataCell stopsCell\\"><span class=\\"val\\">Direkt</span><span class=\\"sub\\">1 saat 30 dakika</span></div><div class=\\"cell priceCell\\"><div class=\\"price\\"><div class=\\"dataCell\\"><span class=\\"val\\">66.99 TL</span><span class=\\"sub\\">Pegasus Airlines</span></div><div class=\\"buttonContainer\\"><button class=\\"button viewButton\\" type=\\"button\\"><span class=\\"text\\">İncele</span></button></div></div></div></div></div>"`;
